app/models/talk.rb

```diff
@@ -41,26 +52,56 @@ class Talk < ApplicationRecord
   # delegates
   delegate :name, to: :event, prefix: true, allow_nil: true
 
+  before_save :compute_embedding, if: :must_compute_embedding?
+
   # search
-  meilisearch do
+  meilisearch primary_key: :id, enqueue: true, raise_on_failure: Rails.env.development? do
     attribute :title
     attribute :description
     attribute :slug
-    attribute :video_id
+    # ⚠️ This `video_id` attribute makes indexing (silently) fail with v1.3.2 of meilisearch. Error message from meilisearch (GET /tasks):
+    #   "The primary key inference failed as the engine found 2 fields ending with `id` in their names: 'id' and 'video_id'. Please specify the primary key manually using the `primaryKey` query parameter"
+    # Adding a custom primary_key: :id above didn't make any difference, so removing this attribute for now.
+    # attribute :video_id
     attribute :video_provider
     attribute :thumbnail_sm
     attribute :thumbnail_md
     attribute :thumbnail_lg
     attribute :speakers do
       speakers.pluck(:name)
     end
+    # ⚠️ This must return nil and not an empty array if no vector is available.
+    # Otherwise all other indexing tasks with non-zero vector arrays will silently fail, since the engine will expect all vectors to have the same length.
+    attribute :_vectors
     searchable_attributes [:title, :description]
     sortable_attributes [:title]
+    filterable_attributes [:id]
 
     attributes_to_highlight ["*"]
   end
 
-  meilisearch enqueue: true
+  # Recomputes embedding for all talks that don't have one yet.
+  def self.reembed!(sleep_interval: 2.seconds, limit: nil)
+    # required for querying vectors (not indexing)
+    MeiliSearch::Rails.client.http_patch "/experimental-features", {vectorStore: true}
```

Comment:
@Kerollmops could this be auto-enabled in some way if we start to send vectors?

Reply:
Unfortunately not. You must enable it by hand 😞